packages/frontend/core/src/modules/telemetry/services/telemetry.ts

```diff
@@ -25,14 +18,19 @@ import { WorkbenchService } from '../../workbench';
 export class TelemetryService extends Service {
   private prevQuota: NonNullable<QuotaQuery['currentUser']>['quota'] | null =
     null;
+  private readonly disposables: (() => void)[] = [];
 
-  constructor(private readonly auth: AuthService) {
+  constructor(
+    private readonly auth: AuthService,
+    private readonly globalContextService: GlobalContextService
+  ) {
     super();
   }
 
   onApplicationStart() {
     const account = this.auth.session.account$.value;
     this.updateIdentity(account);
+    this.registerMiddlewares();
```

Comment:
I tried to use `ApplicationStarted` event for syncing tabs, however it seems this requires the service to be instantiated before this event has been emitted.

Reply:
This is not the case as long as the service is in the root scope

Reply:
I guess it the service must be eagarly created